app/models/friendship.rb

```diff
@@ -1,4 +1,26 @@
 class Friendship < ApplicationRecord
-  belongs_to :friend_a, class_name: :User
-  belongs_to: :friend_b, class_name: :User
+  after_create :create_inverse_relationship
+  after_destroy :destroy_inverse_relationship
+
+  belongs_to :user
+  belongs_to :friend, class_name: 'User'
+
+  validates :user, presence: true
+  validates :friend, presence: true, uniqueness: { scope: :user }
+  validate :not_self
+
+  private
+
+  def create_inverse_relationship
+    friend.friendships.create(friend: user)
```

Comment:
Can we handle this without creating 2 Friendship objects?

Reply:
Will work on this issue in latest branch